packages/core/src/internal/motion/motion.service.spec.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { ClarityMotion } from './motion.service.js';
+
+describe('ClarityMotion service: ', () => {
+  const setConfig = [
+    {
+      target: '.test',
+      animation: [{ opacity: 0 }, { opacity: 1 }],
+      options: {
+        duration: 350,
+        easing: 'ease-out',
+      },
+    },
+  ];
+
+  ClarityMotion.add('test', setConfig);
+
+  describe('get()', () => {
+    it('returns animation configs that were added to the service', () => {
+      const test = ClarityMotion.get('test');
+      expect(test).toEqual(setConfig);
+    });
+
+    it('returns empty if animation config is not found', () => {
+      const nope = ClarityMotion.get('asdfaeawfwefasddf');
+      expect(nope).toEqual([]);
+    });
+  });
+
+  describe('has()', () => {
+    it('returns true if animation configs were added to the service', () => {
+      const test = ClarityMotion.has('test');
+      expect(test).toEqual(true);
+    });
+
+    it('returns false if animation config is not found', () => {
+      const nope = ClarityMotion.has('asdfaeawfwefasddf');
+      expect(nope).toEqual(false);
+    });
+  });
+
+  describe('add()', () => {
+    // add basic functionality tested in get()
+    it('early returns if passed a falsy animation name', () => {
+      spyOn(ClarityMotion, 'add');
+      ClarityMotion.add(null, setConfig);
+      ClarityMotion.add(void 0, setConfig);
+      ClarityMotion.add('', setConfig);
+      expect(ClarityMotion.add).not.toThrowError();
```

Comment:
Is testing that it doesn't throw an error the same as an early return? 
Could a check for a return of `undefined` also be used here?
Same comment for the test below.

```
const undf =  ClarityMotion.add(null, setConfig);
expect(undf).toBeUndefined();
```

Reply:
The problem there is it returns undefined either way. So testing to see if it returns undefined would pass even if it threw an error.